### PR TITLE
WRQ-15921: Fix `Popup` to not focus the element of the popup container when popup is hidden

### PR DESCRIPTION
**Pull request by juwonjeong**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
In FixedPopupPanels when popup is closed, focus does not restore into the background component. It restore focus to the element in the popup panels.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
If the FixedPopupPanels is open when the sample is first rendered, the activator is set to null.

In this case, to restore the focus spotlight module use `getTargetByContainer()`.
In `getTargetByContainer()`, they use `getContainerToSearch()` to get a list of containers to search the focus target, and call `getContainerFocusTarget()` for each container in this list to find the element to focus on. 

Since the popup container is the last container, it is included in the search list and even has the highest priority.

So to fix this issue, we need to prevent popup container to be included in the result of the `getContainerToSearch()`
function. So I added logic to filter popup container with disabling container's selector. 

I fixed this issue by disabling lastContainerId when activator is null.
### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-15921

### CODE REVIEW

**Comment by 0x64** [comment]
The bug occurs when `activator` is not correctly set. So, I prefer to show up the issue condition directly. I'm not sure that what we're supposed to do when `current` does not exist, but it looks okay to handle this case with the same manner. My idea looks like the below. (Just idea, not tested fully.)

>		if (!current || (containerNode && containerNode.contains(current))) {
>			// attempt to set focus to the activator, if available
>			if (!Spotlight.isPaused()) {
>				if (activator) {
>					if (!Spotlight.focus(activator)) {
>						Spotlight.focus();
>					}
>				} else {
>					Spotlight.disableSelector(lastContainerId);
>					Spotlight.focus();
>					Spotlight.enableSelector(lastContainerId);
>				}
>			}
>		}

**Comment by 0x64** [comment]
The bug occurs when `activator` is not correctly set. So, I prefer to show up the issue condition directly.

>		if (!current || (containerNode && containerNode.contains(current))) {
>			// attempt to set focus to the activator, if available
>			if (!Spotlight.isPaused()) {
>				if (activator) {
>					if (!Spotlight.focus(activator)) {
>						Spotlight.focus();
>					}
>				} else {
>					Spotlight.disableSelector(lastContainerId);
>					Spotlight.focus();
>					Spotlight.enableSelector(lastContainerId);
>				}
>			}
>		}

**Comment by juwonjeong** [comment]
> The bug occurs when `activator` is not correctly set. So, I prefer to show up the issue condition directly. I'm not sure that what we're supposed to do when `current` does not exist, but it looks okay to handle this case with the same manner. My idea looks like the below. (Just idea, not tested fully.)
> 
> > ```
> >   if (!current || (containerNode && containerNode.contains(current))) {
> >   	// attempt to set focus to the activator, if available
> >   	if (!Spotlight.isPaused()) {
> >   		if (activator) {
> >   			if (!Spotlight.focus(activator)) {
> >   				Spotlight.focus();
> >   			}
> >   		} else {
> >   			Spotlight.disableSelector(lastContainerId);
> >   			Spotlight.focus();
> >   			Spotlight.enableSelector(lastContainerId);
> >   		}
> >   	}
> >   }
> > ```

Thank you for the review. I agree with your comment that activator should be used as a condition. I tested it and it seems to work fine.

**Review by 0x64 — APPROVED** [review]
Looks good to me!

File: Popup/Popup.js [hunk 1]
```diff
@@ -611,15 +612,24 @@ class Popup extends Component {
 
 		const current = Spotlight.getCurrent();
 		const containerNode = getContainerNode(this.state.containerId);
+		const lastContainerId = getLastContainer();
```
Comment (MikyungKim) [inline Popup/Popup.js:615]: Do we need to know the last container id? Or can we use the containerId of this Popup(`this.state.containerId`)?

File: tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('FixedPopupPanels', function () {
+	const Interface = Page.fixedPopupPanels;
+
+	beforeEach(async function () {
+		await Page.open('WithPopupOpen');
+	});
+
+	describe('5-way', function () {
+
```
Comment (MikyungKim) [inline tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js:11]:
```suggestion
```

File: tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const Page = require('./FixedPopupPanelsWithPopupOpenPage');
+
+describe('FixedPopupPanels', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js:3]:
```suggestion
describe('FixedPopupPanelsWithPopupOpen', function () {
```

File: Popup/Popup.js (hunk 1 above)
Comment (juwonjeong) [reply Popup/Popup.js:615]: As I checked, we need to know the last container id.  For example, in the sample code lastContainerId is `panel-container5` and this.state.containerId returns `container-2`. So when we disable this.state.containerId(=`container-2`), spotlight logic still call `getContainerFocusTarget()` with `panel-container5` which will reproduce the issue.

File: tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js (hunk 2 above)
Comment (juwonjeong) [reply tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js]: It's fixed.

File: tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js (hunk 3 above)
Comment (juwonjeong) [reply tests/ui/specs/FixedPopupPanels/WithPopupOpen/FixedPopupPanelsWithPopupOpen-specs.js]: It's fixed.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**